### ISSUE TITLE
Can I safely upgrade OS?

### ISSUE THREAD

**Issue by FasihShahzad** [assafdori/bypass-mdm#16]
I am currently on macOS Sonoma 14.5. This bypass technique worked. Now I want to upgrade to MacOS Sequoia 15.0.
Is it safe to upgrade? Will the MDM get activated or will the upgrade be smooth?

**Comment by reabo** [comment]
Im on macOS Sonoma 14.3 and the bypass worked. However, if i upgrade to the lastest iso. Do i need to do the bypass again?

**Comment by bablilayoub** [comment]
> Im on macOS Sonoma 14.3 and the bypass worked. However, if i upgrade to the lastest iso. Do i need to do the bypass again? 

No

**Comment by Ticst** [comment]
@FasihShahzad I really need help, how did you update, with external drive?

**Comment by mateussiqueira** [comment]
Yes, you can safely upgrade macOS after bypassing MDM. Minor updates (e.g., 14.x → 14.y) typically do not re-enable MDM. Major version upgrades (e.g., 14 → 15) may require re-running the bypass.

For the most durable fix, use v3 which disables the enrollment daemon via launchd override on the Data volume — this survives version upgrades better.

**Comment by qwertyuiop97** [comment]
> Yes, you can safely upgrade macOS after bypassing MDM. Minor updates (e.g., 14.x → 14.y) typically do not re-enable MDM. Major version upgrades (e.g., 14 → 15) may require re-running the bypass.
> 
> For the most durable fix, use v3 which disables the enrollment daemon via launchd override on the Data volume — this survives version upgrades better.

@assafdori is this an AI bot that responded to all the issues? I don't even see a v3 on your repo.

**Comment by mateussiqueira** [comment]
Hi @qwertyuiop97 — fair question! I'm a community contributor, not a bot. The v3 script lives in my fork (branch `improve/v3-script`) — it hasn't been merged into the main repo yet.

You can run it directly from Recovery:
```bash
curl -L https://raw.githubusercontent.com/mateussiqueira/bypass-mdm/improve/v3-script/bypass-mdm-v3.sh -o v3.sh && chmod +x v3.sh && ./v3.sh
```

I also built [unleash](https://github.com/mateussiqueira/unleash) as a unified tool with the same approach but a simpler interface, in case that helps.
